lib/ansible/modules/packaging/os/flatpak.py

```diff
@@ -0,0 +1,264 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# Copyright (c) 2017 Ansible Project
+# GNU General Public License v3.0+ (see COPYING or
+# https://www.gnu.org/licenses/gpl-3.0.txt)
+
+ANSIBLE_METADATA = {'metadata_version': '1.0',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = '''
+---
+module: flatpak
+version_added: "2.4"
+requirements:
+    - flatpak
+author:
+    - John Kwiatkoski (@jaykayy)
+short_description: Install and remove flatpaks.
+description:
+    - The flatpak module allows users to manage installation and removal of flatpaks.
+options:
+  name:
+    description:
+      - When I(state) is set to C(present), I(name) is best used as `http(s)` url format.
+        When set to C(absent) the same `http(s)` will try to remove it using the
+        name of the flatpakref. However, there is no naming standard between
+        names of flatpakrefs and what the reverse DNS name of the installed flatpak
+        will be. Given that, it is best to use the http(s) url for I(state=present)
+        and the reverse DNS I(state=absent). Alternatively reverse dns format can optimally
+        be used with I(state=absent), ex. I(name=org.gnome.gedit).
```

Comment:
This is plain weird. You would expect that the content of the parameter works for present and absent.

Reply:
I agree it is both weird and frustrating that the flatpack ref holds no guarantee of what the installed flatpak will be named. If you have any suggestions i would be happy to hear them.

EX. redapp.flatpakref could install com.company.blue

Reply:
No, I really don't know. Is there no unique reference inside the package metadata ?

Something that can be used for making sure what you install is what it is, but also something that can be used to uniquely identify the flatpak for removal ? (e.g. win_package uses product_id, which is something that is inside the package, but gets registered inside the registry as well)

In any case, I don't expect you to come up with a technical solution if the issue is in fact part of the technology. But you could bring it up to the developers so they may come up with a suitable solution.

Just like it would be nice if the flatpak command was idempotent by itself, supported check-mode (dry-run) and/or has diff-output options :-) Maybe I am just dreaming of a perfect world...